peachjam/views/generic_views.py

```diff
@@ -175,16 +208,23 @@ def get_queryset(self):
     def filter_queryset(self, qs, filter_q=False):
         return self.form.filter_queryset(qs, filter_q=filter_q)
 
+    def doc_count(self, context):
+        count = cache.get(f"{self.cache_key_prefix}_doc_count")
```

Comment:
What I mean is,

1. if we're using the paginator, use its count
2. otherwise, do our local counting, which involves checking the cache, and updating it if necessary